scripts/lib/python/cve/nvd_cve.py

```diff
@@ -0,0 +1,295 @@
+        cvssv3 = elt['cve']['metrics']['cvssMetricV30'][0]['cvssData']['baseScore']
+    except KeyError:
+        pass
+    try:
+        accessVector = accessVector or elt['cve']['metrics']['cvssMetricV31'][0]['cvssData']['attackVector']
+        vectorString = vectorString or elt['cve']['metrics']['cvssMetricV31'][0]['cvssData']['vectorString']
+        cvssv3 = cvssv3 or elt['cve']['metrics']['cvssMetricV31'][0]['cvssData']['baseScore']
+    except KeyError:
+        pass
+    accessVector = accessVector or "UNKNOWN"
+    vectorString = vectorString or "UNKNOWN"
+    cvssv3 = cvssv3 or 0.0
+
+    conn.execute("insert or replace into NVD values (?, ?, ?, ?, ?, ?, ?, ?)",
+                [cveId, vulnStatus, cveDesc, cvssv2, cvssv3, date, accessVector, vectorString]).close()
+
+    try:
+        for config in elt['cve']['configurations']:
+            # This is suboptimal as it doesn't handle AND/OR and negate, but is better than nothing
+            for node in config["nodes"]:
+                parse_node_and_insert(conn, node, cveId)
+    except KeyError:
+            logger.debug("CVE %s has no configurations" % cveId)
+
+def nvd_request_next(url, api_key, args):
+    """
+    Request next part of the NVD dabase
+    """
+
+    request = urllib.request.Request(url + "?" + urllib.parse.urlencode(args))
+    if api_key:
+        request.add_header("apiKey", api_key)
+    logger.debug(f"Requesting {request.full_url}")
+
+    for attempt in range(5):
+        try:
+            r = urllib.request.urlopen(request)
+
+            if (r.headers['content-encoding'] == 'gzip'):
+                buf = r.read()
+                raw_data = gzip.decompress(buf)
+            else:
+                raw_data = r.read().decode("utf-8")
+
+            r.close()
+        except Exception as e:
+            logger.debug(f"CVE database: received error ({e}), retrying")
+            time.sleep(6)
+            pass
+        else:
+            return raw_data
+    else:
+        # We failed at all attempts
+        return None
+
+def fetch_all_cves(db_file, conn, last_modified, api_key):
+
+    index = 0
+    url = NVDCVE_URL
+
+    req_args = {}
+
+    if last_modified is not None:
+        req_args["lastModStartDate"] = last_modified
+        req_args["lastModEndDate"] = datetime.datetime.now().isoformat()
+
+    # Recommended by NVD
+    sleep_time = 6
+    if api_key:
+        sleep_time = 2
+
+    with open(db_file, 'a') as cve_f:
+        while True:
+            logger.debug("Updating entries")
+
+            req_args["startIndex"] = index
+
+            raw_data = nvd_request_next(url, api_key, req_args)
+            if raw_data is None:
+                return False
+
+            data = json.loads(raw_data)
+
+            index = data["startIndex"]
+            total = data["totalResults"]
+            per_page = data["resultsPerPage"]
+            logger.info(f"Got {per_page} entries")
```

Comment:
This should be `logger.debug` if we treat it as a verbose output.

Reply:
fixed.